source/_docs/terminus/scripting.md

```diff
@@ -0,0 +1,44 @@
+---
+title: Terminus Manual
+subtitle: Scripting Terminus
+description: Automate your workflow with Terminus.
+terminuspage: true
+type: terminuspage
+layout: terminuspage
+nexturl: terminus/plugins/
+previousurl: terminus/commands/
+permalink: docs/terminus/:basename/
+image: terminus-thumbLarge
+searchboost: 100
+---
+
+While Terminus makes it easy to manually manipulate the Pantheon environment from your command line, its real value is in scripting. By adding Terminus to your automated tasks, you can truly make Pantheon a part of your standardized workflow.
+
+Consider the repetitive tasks you perform using the Pantheon Dashboard:
+
+ - Can those tasks be executed by Terminus commands?
+ - Can the values required by the commands be derived programmatically?
+
+If so, you consider how you can turn the task into a script.
+
+## Authentication
+
+To streamline your use of Terminus in scripts, make sure that it's authenticated first. The recommended method is to use a [machine token](/docs/terminus/install#machine-token).
+
+## Examples
+
+ - The [Pantheon Example Terminus Auto Update Script](https://github.com/pantheon-systems/example-terminus-auto-update-script){.external} demonstrates how you can use Terminus to automate plugin and theme updating for multiple sites.
+
+
+ - The [Example WordPress Composer](https://github.com/pantheon-systems/example-wordpress-composer){.external} uses Terminus to [deploy staged changes](https://github.com/pantheon-systems/example-wordpress-composer/blob/master/.circleci/deploy}to-pantheon.sh){.external} to multidev environments.
+
+
+## Bash Variables
+
+One of the ways Terminus can be used in script is the generation of environment variables. In the example below, we use the output of `terminus multidev:list` to create an environment variable with all our multidev environments:
+
+```bash
+PANTHEON_MULTIDEV_LIST="$(terminus multidev:list -n ${TERMINUS_SITE} --format=list --field=Name)"
```

Comment:
We should probably expand on this. _Why_ would you want to store a list of multidev environments in a bash variable? We need to show the value here - being able to leverage that list to complete a larger task.

I think looping through multiple sites would make an even better example. Here is an example script looping through all sites in an org and making a backup on the live environment. It could be extended to download the backup and store it offsite.

```bash
#!/bin/bash

# Exit on error
set -ex

# Stash list of all Pantheon sites in the org
PANTHEON_SITES="$(terminus org:site:list -n ${ORG_UUID} --format=list --field=Name)"

# Loop through each site in the list
while read -r PANTHEON_SITE_NAME; do
    # Create a backup of the live environment of the site
    echo -e "Creating a backup of the live environment for the site '$PANTHEON_SITE_NAME'...\n"
    terminus backup:create $PANTHEON_SITE_NAME.live
done <<< "$PANTHEON_SITES"
```

This example assumes `ORG_UUID` is set to an org UUID retrieved from `terminus org:list`

Reply:
I added that line as an example of how to set the variable as folks might not know how to set a bash variable. I think it is still worth noting in the doc since if line 7 isn't updated the script will fail.